### PR TITLE
Add a databricks workflow taskgroup that converts Airflow tasks into workflow tasks

### PR DESCRIPTION
**Pull request by dimberman**
This pull request creates a new Airflow task group called the DatabricksWorkflowTaskGroup. Databricks workflows are a popular new Databricks product that allows users to optimize their tasks for use on Databricks "job" clusters. The DatabricksWorkflowTaskGroup reads all tasks in the task group and submits them to Databricks as one job. Tasks will need to implement the "convert_to_databricks_workflow_task" function to work in a Databricks task group.

### CODE REVIEW

File: astronomer/providers/databricks/task_group/workflow_taskgroup.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+gfrom __future__ import annotations
```
Comment (kaxil) [inline astronomer/providers/databricks/task_group/workflow_taskgroup.py:1]:
```suggestion
from __future__ import annotations
```

File: astronomer/providers/databricks/task_group/workflow_taskgroup.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+from logging import Logger
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    pass
```
Comment (kaxil) [inline astronomer/providers/databricks/task_group/workflow_taskgroup.py:7]: ?

File: astronomer/providers/databricks/task_group/workflow_taskgroup.py (hunk 1 above)
Comment (kaxil) [inline astronomer/providers/databricks/task_group/workflow_taskgroup.py:1]: We should remove `_taskgroup` from the file name

File: astronomer/providers/databricks/operators/databricks.py [hunk 3]
```diff
@@ -387,3 +387,91 @@ def execute_complete(
 
         self.log.info("%s completed successfully.", self.task_id)
         return None
+
+
+class DatabricksNotebookOperator(BaseOperator):
```
[the remark this answers is not in the document]
Comment (dimberman) [reply astronomer/providers/databricks/operators/databricks.py]: Fixed

File: tests/databricks/taskgroup/test_workflow.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+
+# TODO should we delete this test? It doesn't seem like there are integration tests in
+# this library
+
+# actual_databricks_job = {
+#     "job_id": 183826003004249,
+#     "creator_user_name": "daniel@astronomer.io",
+#     "run_as_user_name": "daniel@astronomer.io",
+#     "run_as_owner": True,
```
Comment (rajaths010494) [inline tests/databricks/taskgroup/test_workflow.py:144]: Can we remove this comment code if not needed.

File: astronomer/providers/databricks/example_dags/example_databricks_workflow.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+"""Example DAG for using the DatabricksWorkflowTaskGroup and DatabricksNotebookOperator."""
```
Comment (rajaths010494) [inline astronomer/providers/databricks/example_dags/example_databricks_workflow.py:1]: Is this example dag self sufficient?

Comment (dimberman) [reply astronomer/providers/databricks/example_dags/example_databricks_workflow.py:1]: @rajaths010494 I think it is sufficient as currently this taskgroup only works with this one operator. Is there something you would like to see to make it more descriptive?

Comment (rajaths010494) [reply astronomer/providers/databricks/example_dags/example_databricks_workflow.py:1]: Not descriptive wanted to make sure this example dag runs fine end to end with out having to setup by us for any infra.
If it's already a self-sufficient dag then good.

File: astronomer/providers/databricks/operators/notebook.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsCreate
+        :param databricks_conn_id: the connection id to use to connect to Databricks
+        :param notebook_params: the parameters to pass to the notebook
+        :param kwargs:
+        """
+        self.notebook_path = notebook_path
+        self.source = source
+        self.notebook_params = notebook_params or {}
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_run_id = ""
```
Comment (phanikumv) [inline astronomer/providers/databricks/operators/notebook.py:94]: can we have a databricks job link on the UI for the users to quickly see the job status from airflow, similar to how google bigquery operators have it?

**Comment by phanikumv** [comment]
Closing as this PR will be created in another project - as discussed with @dimberman

**Pull request closed without merging**